src/index.ts

```diff
@@ -42,8 +42,8 @@ function esBuildSourceMapOptions(tsConfig: TSConfig) {
     return false;
   }
 
-  if ((inlineSources && sourceMap) || sourceMap) {
-    return "external";
+  if (sourceMap) {
```

Comment:
maybe move this condition after the `if(inlineSourceMap)`

Reply:
Done :)